server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -661,6 +661,17 @@ private static void parseNonDynamicArray(ParseContext context, ObjectMapper mapp
         throws IOException {
         XContentParser parser = context.parser();
         XContentParser.Token token;
+        // block array values for composite index fields
+        if (context.indexSettings().isCompositeIndex() && context.mapperService().isFieldPartOfCompositeIndex(arrayFieldName)) {
```

Comment:
Please add a test for this

Reply:
Same , StarTreeMapperIT has all cases covered

```
    public void testCompositeIndexWithArraysInCompositeField() throws IOException 
    public void testCompositeIndexWithArraysInNonCompositeField() throws IOException {
```

Reply:
Got it. I meant from UT perspective. Lets see if we see issues with coverage and we may have to add UTs as well (I don't think ITs are accounted for in coverage)